test/brsTypes/components/RoSGNode.test.js

```diff
@@ -278,4 +278,333 @@ describe("RoSGNode", () => {
+                ]);
+                let child1 = new RoSGNode([
+                    { name: new BrsString("child"), value: new BrsString("2") },
+                ]);
+                let child2 = new RoSGNode([
+                    { name: new BrsString("child"), value: new BrsString("3") },
+                ]);
+                let child3 = new RoSGNode([
+                    { name: new BrsString("child"), value: new BrsString("4") },
+                ]);
+
+                let getChildren = parent.getMethod("getchildren");
+                let getChildCount = parent.getMethod("getchildcount");
+                let appendChild = parent.getMethod("appendchild");
+
+                appendChild.call(interpreter, child1);
+                appendChild.call(interpreter, child2);
+                appendChild.call(interpreter, child3);
+
+                let result = getChildren.call(interpreter, new Int32(-1), new Int32(0));
+                let childCount = getChildCount.call(interpreter);
+                expect(childCount).toEqual(new Int32(3));
+                expect(result).toBeInstanceOf(RoArray);
+                expect(result.elements).toEqual([child1, child2, child3]);
+            });
+
+            it("get subset of children of node", () => {
+                let parent = new RoSGNode([
+                    { name: new BrsString("parent"), value: new BrsString("1") },
+                ]);
+                let child1 = new RoSGNode([
+                    { name: new BrsString("child"), value: new BrsString("2") },
+                ]);
+                let child2 = new RoSGNode([
+                    { name: new BrsString("child"), value: new BrsString("3") },
+                ]);
+                let child3 = new RoSGNode([
+                    { name: new BrsString("child"), value: new BrsString("4") },
+                ]);
+                let child4 = new RoSGNode([
+                    { name: new BrsString("child"), value: new BrsString("5") },
+                ]);
+
+                let getChildren = parent.getMethod("getchildren");
+                let getChildCount = parent.getMethod("getchildcount");
+                let appendChild = parent.getMethod("appendchild");
+
+                appendChild.call(interpreter, child1);
+                appendChild.call(interpreter, child2);
+                appendChild.call(interpreter, child3);
+                appendChild.call(interpreter, child4);
+
+                // get last 2 children
+                let result = getChildren.call(interpreter, new Int32(2), new Int32(2));
+                let childCount = getChildCount.call(interpreter);
+                expect(childCount).toEqual(new Int32(4));
+                expect(result).toBeInstanceOf(RoArray);
+                expect(result.elements).toEqual([child3, child4]);
+
+                // pass in more children count than there are children, should just return 2 with offset of 2
```

Comment:
There seem to be a few tests cases in this single `it`.  For future PRs, consider splitting those into separate test cases — possibly in a `describe()` with a `beforeEach()` to initialize the parent and children!

Reply:
good call!  will break it up :)

Reply:
Oh I only meant for the next PR, you didn't have to do it for this one!  But I appreciate it 😍